wrappers/src/query_cs.cpp

```diff
@@ -391,4 +392,38 @@ REALM_EXPORT Results* query_create_sorted_results(Query * query_ptr, SharedRealm
     });
 }
 
+
+#pragma mark  PrimaryKey Searches
+Row* row_for_primarykey(Table* table_ptr, ObjectSchema* object_schema, std::function<size_t(Table*, size_t columnIndex)> finder, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        size_t columnIndex = realm::npos;
+        if (object_schema->primary_key.size() > 0)
+            columnIndex = table_ptr->get_column_index(object_schema->primary_key);
+        if (columnIndex == realm::npos)
+            throw PrimaryKeyNotDeclaredException(object_schema->name);
```

Comment:
I saw `primary_key_property` is implemented calling `property_for_name` and if the user hasn't specified one, it will loop through all defined properties before failing. Quickly checking if the name is blank seemed a better way to go. If you insist on doing the alternative, how about enhancing `property_for_name` to check for empty names?